File: OOP_Variable/속성_4.rb
class C
    attr_reader :value # 읽는 속성
    attr_writer :value # 쓰는 속성
    #attr_accessor :vlaue # 읽고 쓰기가 가능해짐 (reader, writer 동시) 
    # ruby에서는 attribute. value를 외부에서 읽고 쓰기가 가능한 값으로 만듦
    def initialize(v)
        @value = v        
    end
    def show()
        p @value        
    end

end

c1 = C.new(10)
c1.value = 20
p c1.value # 내부가 아닌 외부에서 접근하는 것을 속성이라고 함
c1.show

File: OOP_Variable/속성_4.py
class C(object):
    def __init__(self, v): 
        self.__value = v # 변수 앞에 __ 가 붙으면 인스턴스 외부에서 접근하는 게 불가능해짐
    def show(self):
        print(self.__value)

c1 = C(10)
# print(c1.__value) # 외부에서 접근이 안 됨
c1.show() # 메소드로서는 접근 가능해짐. 메소드 안에 print + 변수값이 있기 때문(?)